File: lib/wordfilter.rb
require 'json'

module Wordfilter
	BadWordsFileName = File.dirname(__FILE__) + "/badwords.json"
	@@blacklist = nil
	
	def self.init_first_time
		return if(!@@blacklist.nil?)
		self.init
	end
	
	def self.init
		badwords_file = File.read(BadWordsFileName)
		@@blacklist = JSON.parse(badwords_file)
		@@blacklist.map!(&:downcase)
		return
	end
	
	def self.blacklisted? string
		self.init_first_time
		
		string_to_test = string.downcase
		
		@@blacklist.each{|word|
			return true if string_to_test.include? word
		}
		
		return false
	end
	
	def self.add_words words
		self.init_first_time
		words.map!(&:downcase)
		@@blacklist += words
		return
	end

	def self.remove_word word
		self.init_first_time
		@@blacklist.delete word.downcase
		return
	end
	
	def self.clear_list
		@@blacklist = []
		return
	end
end

File: lib/wordfilter.py
import os
import json


class Wordfilter:
    def __init__(self):
        # json is in same directory as this class, given by __location__.
        __location__ = os.path.realpath(
            os.path.join(os.getcwd(), os.path.dirname(__file__)))
        with open(os.path.join(__location__, 'badwords.json')) as f:
            self.blacklist = json.loads(f.read())

    def blacklisted(self, string):
        test_string = string.lower()
        for badword in self.blacklist:
            if badword in test_string:
                return True

        return False

    def addWords(self, words):
        self.blacklist.extend([word.lower() for word in words])

    def removeWord(self, word):
        self.blacklist = [x for x in self.blacklist if x != word]

    def clearList(self):
        self.blacklist = []
